server/tserver/src/main/java/org/apache/accumulo/tserver/TabletServer.java

```diff
@@ -1812,7 +1884,26 @@ public TSummaries getSummariesFromFiles(TInfo tinfo, TCredentials credentials, T
       BlockCache summaryCache = resourceManager.getSummaryCache();
       BlockCache indexCache = resourceManager.getIndexCache();
       FileSystemResolver volMgr = p -> fs.getVolumeByPath(p).getFileSystem();
-      return new Gatherer(TabletServer.this, request, tableCfg).processFiles(volMgr, files, summaryCache, indexCache, srp).toThrift();
+      Future<SummaryCollection> future = new Gatherer(TabletServer.this, request, tableCfg).processFiles(volMgr, files, summaryCache, indexCache, srp);
+
+      return startSummaryOperation(credentials, future);
+    }
+
+    @Override
+    public TSummaries contiuneGetSummaries(TInfo tinfo, long sessionId) throws NoSuchScanIDException, TException {
+      SummarySession session = (SummarySession) sessionManager.getSession(sessionId);
+      if (session == null) {
+        throw new NoSuchScanIDException();
```

Comment:
That exception is used in a few other places (that are not scan) to indicate no such session id.  I was just following that.  The exception should be named no such session exception.